scripts/run_bon_v2.py

```diff
@@ -0,0 +1,357 @@
+        ids = ids[:10]
+
+    ############################
+    # Load reward model pipeline
+    ############################
+    BATCH_SIZE = args.batch_size
+    logger.info("*** Load reward model ***")
+    reward_pipeline_kwargs = {
+        "batch_size": BATCH_SIZE,  # eval_args.inference_batch_size,
+        "truncation": True,
+        "padding": True,
+        "max_length": 2048,
+        "function_to_apply": "none",  # Compute raw logits
+        "return_token_type_ids": False,
+    }
+    if quantized:
+        model_kwargs = {
+            "load_in_8bit": True,
+            "device_map": {"": current_device},
+            "torch_dtype": torch.float16 if torch.cuda.is_available() else None,
+        }
+    else:
+        model_kwargs = {"device_map": {"": current_device}}
+
+    # if attn_implementation is not specified, this falls back to Hugging Face's default
+    # strategy (which chooses between sdpa and eager depending on pytorch version)
+    if args.attn_implementation:
+        model_kwargs["attn_implementation"] = args.attn_implementation
+
+    model = model_builder(args.model, **model_kwargs, trust_remote_code=trust_remote_code)
+    reward_pipe = pipeline_builder(
+        "text-classification",
+        model=model,
+        tokenizer=tokenizer,
+    )
+
+    ############################
+    # Tokenization settings & dataset preparation
+    ############################
+    # set pad token to eos token if not set
+    if reward_pipe.tokenizer.pad_token_id is None:
+        reward_pipe.model.config.pad_token_id = reward_pipe.tokenizer.eos_token_id
+        reward_pipe.tokenizer.pad_token_id = reward_pipe.tokenizer.eos_token_id
+    # For models whose config did not contains `pad_token_id`
+    if reward_pipe.model.config.pad_token_id is None:
+        reward_pipe.model.config.pad_token_id = reward_pipe.tokenizer.pad_token_id
+
+    # if using fastchat template (no template in tokenizer), make the RM tokenizer output an EOS token
+    if not check_tokenizer_chat_template(tokenizer):
+        reward_pipe.tokenizer.add_eos_token = True
+
+    ############################
+    # Run inference [1/2]" built in transformers
+    ############################
+    # if using HF pipeline, can pass entire dataset and get results
+    # first, handle custom pipelines that we must batch normally
+    if pipeline_builder == pipeline:
+        logger.info("*** Running forward pass via built in pipeline abstraction ***")
+        # this setup can be optimized slightly with one pipeline call
+        # prepare for inference
+        reward_pipe = accelerator.prepare(reward_pipe)
+
+        results = reward_pipe(dataset["text"], **reward_pipeline_kwargs)
+
+        # extract scores from results which is list of dicts, e.g. [{'label': 'LABEL_1', 'score': 0.6826171875},... ]
+        scores = [r["score"] for r in results]
+
+    ############################
+    # Run inference [2/2] custom pipelines
+    ############################
```

Comment:
@saumyamalik @vwxyzjn could be fun to try to add VLLM support RMs. How is that going? Could be in a future PR.